server.go

```diff
@@ -2055,6 +2136,62 @@ func (s *Server) executeDropUserStatement(q *influxql.DropUserStatement, user *U
 	return &Result{Err: s.DeleteUser(q.Name)}
 }
 
+func (s *Server) executeDropSeriesStatement(stmt *influxql.DropSeriesStatement, database string, user *User) *Result {
+	s.mu.RLock()
+
+	// Handle the simple `DROP SERIES <id>` case.
+	if stmt.Source == nil && stmt.Condition == nil {
+		s.mu.RUnlock()
+		return &Result{Err: s.DropSeries(database, stmt.SeriesID)}
+	}
+
+	// Handle the more complicated `DROP SERIES` with sources and/or conditions...
+
+	// Find the database.
+	db := s.databases[database]
+	if db == nil {
+		s.mu.RUnlock()
+		return &Result{Err: ErrDatabaseNotFound}
+	}
+
+	// Get the list of measurements we're interested in.
+	measurements, err := measurementsFromSourceOrDB(stmt.Source, db)
+	if err != nil {
+		s.mu.RUnlock()
+		return &Result{Err: err}
+	}
+
+	s.mu.RUnlock()
+	for _, m := range measurements {
+		var ids seriesIDs
+
+		if stmt.Condition != nil {
+			// Get series IDs that match the WHERE clause.
+			filters := map[uint32]influxql.Expr{}
+			ids, _, _ = m.walkWhereForSeriesIds(stmt.Condition, filters)
+
+			// If no series matched, then go to the next measurement.
+			if len(ids) == 0 {
+				continue
+			}
+
+			// TODO: check return of walkWhereForSeriesIds for fields
+		} else {
+			// No WHERE clause so get all series IDs for this measurement.
+			ids = m.seriesIDs
+		}
+
+		// Delete series by ID.
+		for _, id := range ids {
+			err := s.DropSeries(database, id)
+			if err != nil {
+				return &Result{Err: err}
```

Comment:
`DropSeries` calls `broadCast` which calls `Sync` which needs an `RLock`.  If I don't unlock, we deadlock.